### PR TITLE
Update deps

### PR DESCRIPTION
**Pull request by jacob-delgado**
**What this PR does / why we need it**:

Update various deps in go.mod and minor nits fixed to better follow go conventions. Also handle deprecations for stackdriver

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by lei-tang — APPROVED** [review]

**Comment by kyessenov** [comment]
Needs a PR to handle deprecation.

**Comment by lei-tang** [comment]
@jacob-delgado I added a DNM label so the PR will not be merged automatically before the reviewers' comments are addressed.

File: go.mod [hunk 1]
```diff
@@ -3,27 +3,30 @@ module istio.io/proxy
 go 1.19
 
 require (
-	github.com/cncf/udpa/go v0.0.0-20210930031921-04548b0d99d4
```
Comment (jacob-delgado) [inline go.mod:6]: go get said to use `github.com/cncf/xds/go` instead

**Comment by jacob-delgado** [comment]
> Needs a PR to handle deprecation.

Can I get another review

**Review by zirain — APPROVED** [review]

**Comment by lei-tang** [comment]
> Needs a PR to handle deprecation.

@kyessenov Hi Kuat, has your comment been addressed? If yes, please feel free to remove the DNM label.

**Review by kyessenov — APPROVED** [review]

**Pull request merged**